include/revng/AutoEnforcer/AutoEnforcer.h

```diff
@@ -0,0 +1,183 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <string>
+#include <utility>
+
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringMap.h"
+#include "llvm/Support/Error.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include "revng/AutoEnforcer/AutoEnforcerContext.h"
+#include "revng/AutoEnforcer/AutoEnforcerTarget.h"
+#include "revng/AutoEnforcer/BackingContainerRegistry.h"
+#include "revng/AutoEnforcer/InvalidationEvent.h"
+#include "revng/AutoEnforcer/KindsRegistry.h"
+#include "revng/AutoEnforcer/Pipeline.h"
+#include "revng/Support/Debug.h"
+
+namespace AutoEnforcer {
```

Comment:
Sooooo. How do you like the `AutoEnforcer` name?
I'm up for completely reworking the terminology.

Reply:
lol, i have honestly what may be more correct name

Reply:
Let's discuss this next week in person.
Let's leave this open as a reminder.